android/java/org/chromium/chrome/browser/crypto_wallet/util/AssetUtils.java

```diff
@@ -105,24 +108,55 @@ public static boolean isAuroraAddress(String contractAddress, String chainId) {
                 && chainId.equals(BraveWalletConstants.MAINNET_CHAIN_ID);
     }
 
-    public static String getKeyringForCoinType(int coinType) {
-        String keyring = BraveWalletConstants.DEFAULT_KEYRING_ID;
+    public static String getKeyringForChainId(String chainId) {
```

Comment:
Do you feel it's safe to not throw here? I can simply log an error the same way we do for the method `getKeyringForEthOrSolOnly`. 

I'm already using `getKeyringForEthOrSolOnly` where it's safe, meaning: where we know for sure at 100% it can be **only** a Solana or Ethereum coin. For all the other cases we don't know in advance if it's going to be a Sol, an Eth or a Fil. The biggest problem is that we don't have a coin type but only a chain Id.

I agree that hardcoding the supported chains is suboptimal. What about filing a new issue for having this in the core layer @cypt4?

Reply:
I see what you mean now! 
I'm going to introduce a new parameter (`coinType`) and apply the logic you suggested.